[5.2] Work on closure serialization in Mailer

vlakoff opened the pull request:
I analyzed execution speed of the phpunit test suite and noted some bottlenecks. The biggest one is the closure serialization using the AstAnalyzer, which is very slow even on a very simple closure.

Hence the work here, which of course has room for improvement.
It would be great to be able to easily switch closure serialization mechanism. Maybe someone would want to use the much faster TokenAnalyzer, or try Opis Closure (note it doesn't have the same syntax as SuperClosure).

There is also closure serialization code in the Queue class, maybe there is some DRY possible.

vlakoff committed:
Adjust phpdoc in Mailer class

Message callback is expected to be a string or a closure, stored as a string.

GrahamCampbell commented on tests/Mail/MailMailerTest.php:
```diff
@@ -91,9 +92,9 @@ public function testMailerCanQueueMessagesToItselfOnAnotherQueue()
     {
         list($view, $swift) = $this->getMocks();
-        $mailer = new Illuminate\Mail\Mailer($view, $swift);
+        $mailer = new MailerWithFasterSerializer($view, $swift);
```
Please don't edit this test. Add a new one because you've basically just deleted the original test here.

vlakoff replied:
The point of this PR is precisely to avoid running the default AstAnalyzer (which is incredibly slow even for an empty closure) while touching Mailer the less possible.

vlakoff committed:
Unserialize SerializableClosure how it is expected to be

This way SuperClosure disables error handling during unserialization
then checks the result, in case of errors proper exceptions are thrown.

vlakoff committed:
Isolate closure (un)serialization to dedicated methods

To easen using a different serializer by extending the Mailer class.

vlakoff committed:
Use TokenAnalyzer instead of AstAnalyzer to improve test performances

vlakoff commented:
By the way, I just noted there is no test for the closure unserialization. I had done a syntax error which went unnoticed.

GrahamCampbell commented:
No point then. We want our tests to test the whole thing works as is.

vlakoff commented:
I understand, still a few related, worthful points were raised:
* It would be great to improve modularity of closure serialization.
* Proper unserialization on unqueuing should be tested.

Maybe this could be of some inspiration to some ;)
